sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/HttpClientProvider.java

```diff
@@ -3,17 +3,50 @@
 
 package com.generic.core.http.client;
 
+import com.generic.core.implementation.util.Providers;
+import com.generic.core.util.configuration.Configuration;
+
 /**
  * An interface to be implemented by any generic-core plugin that wishes to provide an alternate {@link HttpClient}
  * implementation.
  */
-@FunctionalInterface
-public interface HttpClientProvider {
+public abstract class HttpClientProvider {
+    final static String NO_DEFAULT_PROVIDER_MESSAGE = "A request was made to load the default HttpClient provider "
+        + "but one could not be found on the classpath. If you are using a dependency manager, consider including "
+        + "a dependency on io.clientcore:http-okhttp. Additionally, refer to "
+        + "https://aka.ms/azsdk/java/docs/custom-httpclient to learn about writing your own implementation.";
```

Comment:
We need to file an issue to update our links to non-azure docs. Can you please add one?

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/39510